src/cosl/statuses.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""Helper functions for calculating 'pull statuses'.
+
+Ref: https://discourse.charmhub.io/t/its-probably-ok-for-a-unit-to-go-into-error-state/13022
+"""
+
+import logging
+import shutil
+
+import ops
+
+logger = logging.getLogger(__name__)
+
+
+def get_disk_usage_status(location: str, *, threshold: int = 1024**3) -> ops.StatusBase:
+    """Returns a status that matches the disk usage.
+
```

Comment:
Let's add an "Args" section to the docstring to better explain what this function receives.